### PR TITLE
Remove more //base dependencies from //sky/engine

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by abarth** [comment]
@chinmaygarde

File: sky/engine/core/script/dart_init.cc [hunk 1]
```diff
@@ -134,24 +133,32 @@ void IsolateShutdownCallback(void* callback_data) {
...
   }
-  base::ReplaceFirstSubstringAfterOffset(&url, 0, "file:", "");
-  base::FilePath path(url);
-  base::File::Info file_info;
-  if (!base::GetFileInfo(path, &file_info)) {
-    // Assume modified.
+
+  const char* path = source_url + kFileUriPrefixLength;
+  struct stat info;
+  if (HANDLE_EINTR(stat(path, &info)) < 0)
```
Comment (chinmaygarde) [inline sky/engine/core/script/dart_init.cc]: `stat` doesn't return `EINTR`.  You don't need the `HANDLE_EINTR`.

File: sky/engine/core/script/dart_init.cc [hunk 2]
```diff
@@ -405,26 +412,23 @@ void* _DartSymbolLookup(const char* symbol_name) {
     const std::string& aot_snapshot_path = SkySettings::Get().aot_snapshot_path;
     FTL_CHECK(!aot_snapshot_path.empty());
 
-    base::FilePath asset_path =
-        base::FilePath(aot_snapshot_path).Append(symbol_asset.file_name);
-    int64 asset_size;
-    if (!base::GetFileSize(asset_path, &asset_size))
+    std::string asset_path = aot_snapshot_path + "/" + symbol_asset.file_name;
+    struct stat info;
+    if (HANDLE_EINTR(stat(asset_path.c_str(), &info)) < 0)
```
Comment (chinmaygarde) [inline sky/engine/core/script/dart_init.cc]: Same. `stat` doesn't return `EINTR`.

**Comment by chinmaygarde** [comment]
lgtm

File: sky/engine/core/script/dart_init.cc (hunk 2 above)
Comment (abarth) [inline sky/engine/core/script/dart_init.cc]: Done.

File: sky/engine/core/script/dart_init.cc (hunk 1 above)
Comment (abarth) [inline sky/engine/core/script/dart_init.cc]: Done.

**Pull request merged**